In [1]:
#lets start by reading in the csv file and printing out the first few lines
import pandas as pd

df = pd.read_csv('insurance_claims.csv')


df.head (5)


months_as_customer  age  policy_number policy_bind_date policy_state  \
0                 328   48         521585       2014-10-17           OH   
1                 228   42         342868       2006-06-27           IN   
2                 134   29         687698       2000-09-06           OH   
3                 256   41         227811       1990-05-25           IL   
4                 228   44         367455       2014-06-06           IL   

  policy_csl  policy_deductable  policy_annual_premium  umbrella_limit  \
0    250/500               1000                1406.91               0   
1    250/500               2000                1197.22         5000000   
2    100/300               2000                1413.14         5000000   
3    250/500               2000                1415.74         6000000   
4   500/1000               1000                1583.91         6000000   

   insured_zip  ... police_report_available total_claim_amount injury_claim  \
0       466132  ...                     YES              71610         6510   
1       468176  ...                       ?               5070          780   
2       430632  ...                      NO              34650         7700   
3       608117  ...                      NO              63400         6340   
4       610706  ...                      NO               6500         1300   

  property_claim vehicle_claim  auto_make  auto_model auto_year  \
0          13020         52080       Saab         92x      2004   
1            780          3510   Mercedes        E400      2007   
2           3850         23100      Dodge         RAM      2007   
3           6340         50720  Chevrolet       Tahoe      2014   
4            650          4550     Accura         RSX      2009   

  fraud_reported _c39  
0              Y  NaN  
1              Y  NaN  
2              N  NaN  
3              Y  NaN  
4              N  NaN  

[5 rows x 40 columns]

In [2]:
#lets take a look at the number of rows and columns in the dataframe
df.shape



(1000, 40)

In [3]:
df.columns

Index(['months_as_customer', 'age', 'policy_number', 'policy_bind_date',
       'policy_state', 'policy_csl', 'policy_deductable',
       'policy_annual_premium', 'umbrella_limit', 'insured_zip', 'insured_sex',
       'insured_education_level', 'insured_occupation', 'insured_hobbies',
       'insured_relationship', 'capital-gains', 'capital-loss',
       'incident_date', 'incident_type', 'collision_type', 'incident_severity',
       'authorities_contacted', 'incident_state', 'incident_city',
       'incident_location', 'incident_hour_of_the_day',
       'number_of_vehicles_involved', 'property_damage', 'bodily_injuries',
       'witnesses', 'police_report_available', 'total_claim_amount',
       'injury_claim', 'property_claim', 'vehicle_claim', 'auto_make',
       'auto_model', 'auto_year', 'fraud_reported', '_c39'],
      dtype='object')

# 📊 Insurance Claims Dataset: Field Explanations

"""
🔹 Customer & Policy Information
- months_as_customer: Number of months the person has been a customer
- age: Age of the insured person
- policy_number: Unique identifier for the policy
- policy_bind_date: Date when the policy was issued/bound
- policy_state: State where the policy was issued
- policy_csl: Combined single limit of liability coverage
- policy_deductable: Deductible amount for the policy
- policy_annual_premium: Annual premium paid for the policy
- umbrella_limit: Maximum coverage limit of the umbrella policy

🔹 Insured (Customer) Demographics
- insured_zip: ZIP code of the insured
- insured_sex: Gender of the insured
- insured_education_level: Highest education level of the insured
- insured_occupation: Occupation of the insured
- insured_hobbies: Hobbies of the insured
- insured_relationship: Relationship status of the insured

🔹 Financial Information
- capital-gains: Capital gains reported by the insured
- capital-loss: Capital losses reported by the insured

🔹 Incident Details
- incident_date: Date of the reported incident
- incident_type: Type of incident (e.g., theft, collision, fire)
- collision_type: Type of collision (if applicable)
- incident_severity: Severity of the incident (e.g., minor, major, total loss)
- authorities_contacted: Which authorities were contacted (police, fire dept.)
- incident_state: State where the incident occurred
- incident_city: City where the incident occurred
- incident_location: Specific location/address of the incident
- incident_hour_of_the_day: Hour of the day when the incident occurred

🔹 Incident Outcomes
- number_of_vehicles_involved: How many vehicles were part of the incident
- property_damage: Whether there was property damage (Y/N/Unknown)
- bodily_injuries: Number of bodily injuries reported
- witnesses: Number of witnesses present
- police_report_available: Whether a police report was available (Y/N/Unknown)

🔹 Claim Information
- total_claim_amount: Total claim amount requested/paid
- injury_claim: Claim amount related to injuries
- property_claim: Claim amount related to property damage
- vehicle_claim: Claim amount related to vehicle damage

🔹 Vehicle Information
- auto_make: Make of the insured’s vehicle
- auto_model: Model of the insured’s vehicle
- auto_year: Year of the insured’s vehicle

🔹 Fraud Indicator
- fraud_reported: Whether the claim was flagged/reported as fraud (Y/N)

🔹 Miscellaneous
- _c39: Extra/unidentified column (may be an artifact from CSV export)
"""


In [4]:
#lets get a few summary statistics of the data
df.describe()

months_as_customer          age  policy_number  policy_deductable  \
count         1000.000000  1000.000000    1000.000000        1000.000000   
mean           203.954000    38.948000  546238.648000        1136.000000   
std            115.113174     9.140287  257063.005276         611.864673   
min              0.000000    19.000000  100804.000000         500.000000   
25%            115.750000    32.000000  335980.250000         500.000000   
50%            199.500000    38.000000  533135.000000        1000.000000   
75%            276.250000    44.000000  759099.750000        2000.000000   
max            479.000000    64.000000  999435.000000        2000.000000   

       policy_annual_premium  umbrella_limit    insured_zip  capital-gains  \
count            1000.000000    1.000000e+03    1000.000000    1000.000000   
mean             1256.406150    1.101000e+06  501214.488000   25126.100000   
std               244.167395    2.297407e+06   71701.610941   27872.187708   
min               433.330000   -1.000000e+06  430104.000000       0.000000   
25%              1089.607500    0.000000e+00  448404.500000       0.000000   
50%              1257.200000    0.000000e+00  466445.500000       0.000000   
75%              1415.695000    0.000000e+00  603251.000000   51025.000000   
max              2047.590000    1.000000e+07  620962.000000  100500.000000   

        capital-loss  incident_hour_of_the_day  number_of_vehicles_involved  \
count    1000.000000               1000.000000                   1000.00000   
mean   -26793.700000                 11.644000                      1.83900   
std     28104.096686                  6.951373                      1.01888   
min   -111100.000000                  0.000000                      1.00000   
25%    -51500.000000                  6.000000                      1.00000   
50%    -23250.000000                 12.000000                      1.00000   
75%         0.000000                 17.000000                      3.00000   
max         0.000000                 23.000000                      4.00000   

       bodily_injuries    witnesses  total_claim_amount  injury_claim  \
count      1000.000000  1000.000000          1000.00000   1000.000000   
mean          0.992000     1.487000         52761.94000   7433.420000   
std           0.820127     1.111335         26401.53319   4880.951853   
min           0.000000     0.000000           100.00000      0.000000   
25%           0.000000     1.000000         41812.50000   4295.000000   
50%           1.000000     1.000000         58055.00000   6775.000000   
75%           2.000000     2.000000         70592.50000  11305.000000   
max           2.000000     3.000000        114920.00000  21450.000000   

       property_claim  vehicle_claim    auto_year  _c39  
count     1000.000000    1000.000000  1000.000000   0.0  
mean      7399.570000   37928.950000  2005.103000   NaN  
std       4824.726179   18886.252893     6.015861   NaN  
min          0.000000      70.000000  1995.000000   NaN  
25%       4445.000000   30292.500000  2000.000000   NaN  
50%       6750.000000   42100.000000  2005.000000   NaN  
75%      10885.000000   50822.500000  2010.000000   NaN  
max      23670.000000   79560.000000  2015.000000   NaN

# 📊 Insurance Claims — Summary Statistics (Numeric Features)

Here are the descriptive statistics from the dataset (n = 1000 claims):

---

## 📅 Customer & Policy
| Variable | Count | Mean | Std | Min | 25% | 50% | 75% | Max |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| `months_as_customer` | 1000 | 203.95 | 115.11 | 0 | 115.75 | 199.5 | 276.25 | 479 |
| `age` | 1000 | 38.95 | 9.14 | 19 | 32 | 38 | 44 | 64 |
| `policy_number` | 1000 | 546,238.65 | 257,063.01 | 100,804 | 335,980 | 533,135 | 759,100 | 999,435 |
| `policy_deductable` | 1000 | 1,136.00 | 611.86 | 500 | 500 | 1000 | 2000 | 2000 |
| `policy_annual_premium` | 1000 | 1,256.41 | 244.17 | 433.33 | 1089.61 | 1257.20 | 1415.70 | 2047.59 |
| `umbrella_limit` | 1000 | 1.10e6 | 2.30e6 | -1.00e6 | 0 | 0 | 0 | 1.00e7 |

---

## 🏦 Financials
| Variable | Count | Mean | Std | Min | 25% | 50% | 75% | Max |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| `insured_zip` | 1000 | 501,214.49 | 71,701.61 | 430,104 | 448,405 | 466,446 | 603,251 | 620,962 |
| `capital-gains` | 1000 | 25,126.10 | 27,872.19 | 0 | 0 | 0 | 51,025 | 100,500 |
| `capital-loss` | 1000 | -26,793.70 | 28,104.10 | -111,100 | -51,500 | -23,250 | 0 | 0 |

---

## 🚨 Incident Details
| Variable | Count | Mean | Std | Min | 25% | 50% | 75% | Max |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| `incident_hour_of_the_day` | 1000 | 11.64 | 6.95 | 0 | 6 | 12 | 17 | 23 |
| `number_of_vehicles_involved` | 1000 | 1.84 | 1.02 | 1 | 1 | 1 | 3 | 4 |
| `bodily_injuries` | 1000 | 0.99 | 0.82 | 0 | 0 | 1 | 2 | 2 |
| `witnesses` | 1000 | 1.49 | 1.11 | 0 | 1 | 1 | 2 | 3 |

---

## 💰 Claims
| Variable | Count | Mean | Std | Min | 25% | 50% | 75% | Max |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| `total_claim_amount` | 1000 | 52,761.94 | 26,401.53 | 100 | 41,813 | 58,055 | 70,593 | 114,920 |
| `injury_claim` | 1000 | 7,433.42 | 4,880.95 | 0 | 4,295 | 6,775 | 11,305 | 21,450 |
| `property_claim` | 1000 | 7,399.57 | 4,824.73 | 0 | 4,445 | 6,750 | 10,885 | 23,670 |
| `vehicle_claim` | 1000 | 37,928.95 | 18,886.25 | 70 | 30,293 | 42,100 | 50,823 | 79,560 |

---

## 🚗 Vehicle
| Variable | Count | Mean | Std | Min | 25% | 50% | 75% | Max |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| `auto_year` | 1000 | 2005.10 | 6.02 | 1995 | 2000 | 2005 | 2010 | 2015 |

---

### ⚠️ Notes
- `umbrella_limit` has extreme outliers (min = -1,000,000, max = 10,000,000).  
- `capital-gains` and `capital-loss` are highly skewed (lots of 0s).  
- Claims data (`injury_claim`, `property_claim`, `vehicle_claim`) have heavy right tails — large maximums compared to median.  
- `_c39` column contains no values → candidate for drop.  


In [5]:
#Lets start by dropping any empty columns
import pandas as pd
import numpy as np

#lets drop the _c39 column as it is completely empty
df = df.drop(columns=['_c39'])

# umbrella_limit: inspect and cap / flag
print(df['umbrella_limit'].describe())

count    1.000000e+03
mean     1.101000e+06
std      2.297407e+06
min     -1.000000e+06
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      1.000000e+07
Name: umbrella_limit, dtype: float64


In [6]:
#lets further inspect the umbrella_limit column
print(df['umbrella_limit'].value_counts())



umbrella_limit
 0           798
 6000000      57
 5000000      46
 4000000      39
 7000000      29
 3000000      12
 8000000       8
 9000000       5
 2000000       3
 10000000      2
-1000000       1
Name: count, dtype: int64



Umbrella insurance provides EXTRA liability coverage
on top of existing policies (like auto or homeowners).
It "kicks in" when those underlying policy limits are reached.

🔹 Example:
- Auto insurance liability limit: $250,000
- Accident causes damages: $1,000,000

Auto insurance pays first $250,000.
Umbrella policy pays the remainder up to its umbrella limit.

So, if the umbrella limit is $1,000,000:
- Auto covers: $250,000
- Umbrella covers: $750,000
- Total covered: $1,000,000
- You’re protected from paying out of pocket.

798/1000 rows have umbrella_limit = 0 → most policies have no umbrella coverage.

Many repeated large values (2–10M) suggest a set of standard coverage tiers, not continuous measurements.

One row has -1,000,000 → looks like a sentinel / data error (treat as missing).

Mean and std are dominated by the few large tiers, so raw mean is not representative.

In [7]:
#looks like one record has a negative value, which is impossible
#lets replace it with a NaN value
df.loc[df['umbrella_limit'] < 0, 'umbrella_limit'] = np.nan
# Inspect counts and fraction zeros
print(df['umbrella_limit'].value_counts())


umbrella_limit
0.0           798
6000000.0      57
5000000.0      46
4000000.0      39
7000000.0      29
3000000.0      12
8000000.0       8
9000000.0       5
2000000.0       3
10000000.0      2
Name: count, dtype: int64


In [8]:
#Now lets take a look at the capital-gains column
print(df['capital-gains'].describe())



count      1000.000000
mean      25126.100000
std       27872.187708
min           0.000000
25%           0.000000
50%           0.000000
75%       51025.000000
max      100500.000000
Name: capital-gains, dtype: float64
